Sources/XcodeGenCLI/Commands/GenerateCommand.swift

```diff
@@ -99,37 +99,64 @@ class GenerateCommand: ProjectCommand {
-            success("Created project at \(projectPath)")
-        } catch {
-            throw GenerationError.writingError(error)
-        }
+        var errors: [Error] = []
+        let overallDispatchGroup = DispatchGroup()
+        overallDispatchGroup.enter()
+        let generateAndWriteProject = DispatchWorkItem { [weak self] in
+            var xcodeProject: XcodeProj!
+            let xcodeProjDispatchGroup = DispatchGroup()
+            do {
+                xcodeProjDispatchGroup.enter()
+                let projectGenerator = ProjectGenerator(project: project)
+                try projectGenerator.generateXcodeProject(in: projectDirectory) { (generatedXcodeProj) in
+                    xcodeProject = generatedXcodeProj
+                    xcodeProjDispatchGroup.leave()
+                }
+            } catch {
+                xcodeProjDispatchGroup.leave()
+                errors.append(GenerationError.generationError(error))
+            }
+            xcodeProjDispatchGroup.wait()
 
-        // write cache
-        if let cacheFile = cacheFile {
+            guard errors.isEmpty else {
+                return
+            }
+            // write project
+            self?.info("⚙️  Writing project...")
             do {
-                try cacheFilePath.parent().mkpath()
-                try cacheFilePath.write(cacheFile.string)
+                try fileWriter.writeXcodeProject(xcodeProject, to: projectPath)
+                self?.success("Created project at \(projectPath)")
             } catch {
-                info("Failed to write cache: \(error.localizedDescription)")
+                errors.append(GenerationError.writingError(error))
             }
-        }
 
-        // run post gen command
-        if let command = project.options.postGenCommand {
-            try Task.run(bash: command, directory: projectDirectory.absolute().string)
+            // write cache
+            if let cacheFile = cacheFile {
+                do {
+                    try cacheFilePath.parent().mkpath()
+                    try cacheFilePath.write(cacheFile.string)
+                } catch {
+                    self?.info("Failed to write cache: \(error.localizedDescription)")
+                }
+            }
+
+            // run post gen command
+            if let command = project.options.postGenCommand {
+                do {
+                    try Task.run(bash: command, directory: projectDirectory.absolute().string)
+                } catch let error {
+                    errors.append(error)
+                }
+            }
+        }
+        generateAndWriteProject.notify(queue: .global(qos: .userInitiated)) { [weak self] in
+            if let error = errors.first {
+                self?.stderr.print(error.localizedDescription)
+            }
+            overallDispatchGroup.leave()
         }
+        DispatchQueue.global(qos: .userInitiated).async(execute: generateAndWriteProject)
+        overallDispatchGroup.wait()
```

Comment:
This is the part where I am blocking the main queue, so that the generateCommand gets time to generate all the targets, write the xcodeproj and then finish